org/mozilla/jss/netscape/security/util/Utils.java

```diff
@@ -406,7 +406,16 @@ public static String base64encodeSingleLine(byte[] bytes) {
      */
     public static byte[] base64decode(String string) {
         try {
-            return Base64.getMimeDecoder().decode(string);
+            // Java is particular about its base64. We already used the MIME
+            // decoder as it was most flexible about whitespace. However, it
+            // doesn't understand URL-encoded Base64 (using '-' instead of
+            // '+' and '_' instead of '/'). So, detect those characters and
+            // pass it to the correct decoder.
+            if (string.contains("_") || string.contains("-")) {
+                return Base64.getUrlDecoder().decode(string);
+            } else {
+                return Base64.getMimeDecoder().decode(string);
+            }
```

Comment:
If the string doesn't contain "_" or "-", can it be decoded with any of these methods and will they produce the same results?